plugin/stripcomments/stripcomments.go

```diff
@@ -0,0 +1,105 @@
+package stripcomments
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"slices"
+
+	"github.com/safedep/code/core"
+	"github.com/safedep/dry/log"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type StripCommentsCallback func(f core.File, r io.Reader) error
+
+type stripCommentsPlugin struct {
+	stripCommentsCallback StripCommentsCallback
+}
+
+// Verify contract
+var _ core.TreePlugin = (*stripCommentsPlugin)(nil)
+
+func NewStripCommentsPlugin(stripCommentsCallback StripCommentsCallback) *stripCommentsPlugin {
```

Comment:
I think we should add a documentation (comment) for `NewStripCommentsPlugin` describing this plugin and the algorithm used to strip comments. 

Note: The documentation will eventually be generated by `godoc` and available for users of the library.
https://go.dev/blog/godoc